src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php

```diff
@@ -233,13 +233,13 @@ private function castableToString(array $choices, array &$cache = array())
                 }
 
                 continue;
-            } elseif (!is_scalar($choice)) {
+            } elseif (!is_scalar($choice) && null !== $choice) {
```

Comment:
> This would solve it indeed, but may not result in an expected behaviour when using custom request submission (e.g javascript, api) as one would send an empty value and want it to be mapped to the choice with a null value instead of needing to send '~'.

The counter argument seems wrong. If people used `null`` as a choice, then it used numeric keys for values. So javascript, apis etc would also need to send the numeric keys (depending on the order) and not an empty string.